### PR TITLE
Implement GrandPa forced changes

### PR DESCRIPTION
**Pull request by tomaka**
Doesn't work, but it is some progress.

### CODE REVIEW

**Comment by tomaka** [comment]
cc #515

File: src/chain/blocks_tree/finality.rs [hunk 1]
```diff
@@ -96,21 +96,32 @@ impl<T> NonFinalizedTreeInner<T> {
...
+                //                          authorities                    justification
+                //     finalized              change                          target
+                //         +                    +                               +
+                //         v                    v                               v
+                //         +----------------------------------------------------+
 
                 // Find out the next block height where an authority change will be triggered.
+                let mut overwrite_authorities = None;
                 let earliest_trigger = {
+                    // TODO: this code is clumbsy; rules aren't necessarily respected properly; review and improve
```
Comment (expenses) [inline src/chain/blocks_tree/finality.rs:116]:
```suggestion
                    // TODO: this code is clumsy; rules aren't necessarily respected properly; review and improve
```

File: src/chain/blocks_tree/finality.rs [hunk 2]
```diff
@@ -223,13 +266,33 @@ impl<T> NonFinalizedTreeInner<T> {
                         header::DigestItemRef::GrandpaConsensus(gp) => Some(gp),
                         _ => None,
                     }) {
+                        // TODO: this code is clumbsy; rules aren't necessarily respected properly; review and improve
```
Comment (expenses) [inline src/chain/blocks_tree/finality.rs:269]:
```suggestion
                        // TODO: this code is clumsy; rules aren't necessarily respected properly; review and improve
```

**Comment by tomaka** [comment]
Closing this PR, as all the changes are completely outdated.

**Pull request closed without merging**